pkg/apiserver/rest/webservice/addon.go

```diff
@@ -98,7 +98,7 @@ func (s *addonWebService) GetWebService() *restful.WebService {
 
 func (s *addonWebService) listAddons(req *restful.Request, res *restful.Response) {
 	detailAddons, err := s.addonUsecase.ListAddons(req.Request.Context(), false, req.QueryParameter("registry"), req.QueryParameter("query"))
-	if err != nil {
+	if err != nil && len(detailAddons) == 0 {
```

Comment:
Error handling should not be like that. If `len(detailAddons) == 0` is not an error case, we should not return error